tests/lax_test.py

```diff
@@ -1715,28 +1736,28 @@ def testTransposeAgainstNumpy(self, shape, dtype, perm):
 
   @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_op={}_inshape={}_reducedims={}_initval={}"
-       .format(op.__name__, jtu.format_shape_dtype_string(shape, dtype), dims,
-               init_val),
-       "op": op, "init_val": init_val, "shape": shape, "dtype": dtype, "dims": dims}
-      for init_val, op, types in [
-          (0, lax.add, default_dtypes),
-          (1, lax.mul, default_dtypes),
-          (0, lax.max, all_dtypes), # non-monoidal
-          (-np.inf, lax.max, float_dtypes),
-          (dtypes.iinfo(np.int32).min, lax.max, [np.int32]),
-          (dtypes.iinfo(np.int64).min, lax.max, [np.int64]),
-          (np.inf, lax.min, float_dtypes),
-          (dtypes.iinfo(np.int32).max, lax.min, [np.int32]),
-          (dtypes.iinfo(np.int64).max, lax.min, [np.int64]),
-          (dtypes.iinfo(np.uint32).max, lax.min, [np.uint32]),
-          (dtypes.iinfo(np.uint64).max, lax.min, [np.uint64]),
-      ]
-      for dtype in types
+       .format(rec.op.__name__, jtu.format_shape_dtype_string(shape, dtype), dims,
+               rec.init_val),
+       "op": rec.op, "reference_op": rec.reference_op, "init_val": rec.init_val,
+       "shape": shape, "dtype": dtype, "dims": dims, "primitive": rec.primitive}
+      for rec in LAX_REDUCE_OPS
+      for dtype in rec.dtypes
       for shape, dims in [
           [(3, 4, 5), (0,)], [(3, 4, 5), (1, 2)],
           [(3, 4, 5), (0, 2)], [(3, 4, 5), (0, 1, 2)]
       ]))
-  def testReduce(self, op, init_val, shape, dtype, dims):
+  def testReduce(self, op, reference_op, init_val, shape, dtype, dims, primitive):
+    if not config.x64_enabled and dtype in (np.float64, np.int64, np.uint64):
+      raise SkipTest("x64 mode is disabled.")
+    def reference_fun(operand):
+      if hasattr(reference_op, "reduce"):
+        initial = np.array(init_val, dtype=dtype)
+        result = reference_op.reduce(operand, axis=dims, initial=initial)
+      else:
+        result = reference_op(operand, axis=dims)
+
+      return result.astype(dtype)
+
```

Comment:
One issue just below: because `init_val` is passed via `args_maker`, within `CompileAndCheck` it will be traced and therefore the new reducer primitives will not come into play

Reply:
Thanks, that makes sense. But how is only `bitwise_and` affected?

Reply:
Perhaps something about the monoid assumption is violated?

Reply:
Good idea, but I think would cause tests to fail for all shapes when they do not:

```
[ RUN      ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(0, 1, 2)_initval=-1
[       OK ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(0, 1, 2)_initval=-1
[ RUN      ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(0, 2)_initval=-1
[       OK ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(0, 2)_initval=-1
[ RUN      ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(0,)_initval=-1
[  FAILED  ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(0,)_initval=-1
[ RUN      ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(1, 2)_initval=-1
[       OK ] LaxTest.testReduceBare_op=bitwise_and_inshape=int16[3,4,5]_reducedims=(1, 2)_initval=-1
```

pytest logs are similar for int8/32 as well as uint types.

Reply:
Not necessarily, `lax.reduce` will sometimes work if the arguments do not meet the monoid condition. It's just that it's not guaranteed to return the correct output unless those conditions are met.

Reply:
Alright. I played with the setup a little, and the number of failing tests changes if you change the reduction dimension to `(1,)` or `(2,)`, for example. 

Also, all tests pass when the first call to `_CompileAndCheck` and the call to `_CheckAgainstNumpy` are commented out - the check of the JIT compilation for the operand-only lambda passes. So it might indeed have to do with the lambda's signature. Do you have an idea on how I could proceed?

Reply:
It's not obvious to me where the bug is... I would probably put together a stand-alone reproduction of the error, and start using `jax.make_jaxpr` to inspect the operations that are being generated.